src/cargo/core/resolver/mod.rs

```diff
@@ -316,22 +316,23 @@ fn activate(cx: &mut Context,
                               candidate.summary.package_id().clone());
     }
 
-    if cx.flag_activated(&candidate.summary, method) {
-        return Ok(None);
-    }
+    let activated = cx.flag_activated(&candidate.summary, method);
 
     let candidate = match candidate.replace {
         Some(replace) => {
             cx.resolve_replacements.insert(candidate.summary.package_id().clone(),
                                            replace.package_id().clone());
-            if cx.flag_activated(&replace, method) {
+            if cx.flag_activated(&replace, method) && activated {
```

Comment:
Ah, looks like moving if forward won't work because during replacement we need to activate both the original and the replacement.